core/commands/partials/terraform_configurator.rb

```diff
@@ -98,89 +104,83 @@ def cnf_extra_files(node)
+  def up_machines(nodes)
+    TerraformService.resource_type(@config.provider).and_then do |resource_type|
+      TerraformService.init(@ui, @config.path)
+      target_nodes = TerraformService.nodes_to_resources(nodes, resource_type)
+      @attempts.times do |attempt|
+        @ui.info("Up nodes #{nodes}. Attempt #{attempt + 1}.")
+        destroy_nodes(target_nodes.keys) if @recreate_nodes || attempt.positive?
+        TerraformService.apply(target_nodes.values, @ui, @config.path)
+        TerraformService.running_resources(@ui, @config.path).and_then do |running_resources|
+          target_nodes = target_nodes.reject { |_node, resource| running_resources.include?(resource) }
+        end
+        return Result.ok('') if target_nodes.empty?
+      end
+      Result.error("Error up of machines: #{target_nodes.keys}")
     end
   end
 
   # rubocop:disable Metrics/MethodLength
   def configure_node(node)
-    result = retrieve_network_settings(node).and_then do |node_network|
-      wait_for_node_availability(node, node_network)
-    end.and_then do |node_network|
-      @network_settings.add_network_configuration(node, node_network)
-      if NetworkChecker.resources_available?(@machine_configurator, @network_settings.node_settings(node), @ui)
-        Result.ok('')
-      else
-        Result.error('Network resources not available!')
+    logger = retrieve_logger_for_node
+    result = nil
+    @attempts.times do |attempt|
+      @ui.info("Configure node #{node}. Attempt #{attempt + 1}.")
+      result = retrieve_network_settings(node).and_then do |node_network|
+        wait_for_node_availability(node, node_network)
+      end.and_then do |node_network|
+        @network_settings.add_network_configuration(node, node_network)
+        if NetworkChecker.resources_available?(@machine_configurator, @network_settings.node_settings(node), logger)
+          Result.ok('')
+        else
+          Result.error('Network resources not available!')
+        end
+      end.and_then do
+        configure_with_chef(node, logger)
       end
-    end.and_then do
-      configure_with_chef(node)
-    end
 
-    if result.success?
-      @ui.info("Node '#{node}' has been configured.")
-    else
-      @ui.error("Exception during node configuration: #{result.error}")
+      if result.success?
+        @ui.info("Node '#{node}' has been configured.")
+        break
+      else
+        @ui.error("Exception during node configuration: #{result.error}")
+      end
     end
-    result
+    { node: node, result: result, logger: logger }
   end
   # rubocop:enable Metrics/MethodLength
 
+  # Configure machines via mdbci.
+  #
+  # @param nodes [Array<String>] name of nodes to configure
+  # @return [Result::Base]
+  def configure_nodes(nodes)
+    @ui.info("Configure machines: #{nodes}")
+    configure_results = Workers.map(nodes) { |node| configure_node(node) }
+    configure_results.each { |result| result[:logger].print_to_stdout } if use_log_storage?
```

Comment:
You should go through the results in one iteration. No need to iterate over it two times.